components/script/dom/htmlanchorelement.rs

```diff
@@ -566,6 +568,64 @@ impl Activatable for HTMLAnchorElement {
     }
 }
 
+/// <https://html.spec.whatwg.org/multipage/semantics.html#get-an-element's-target>
+pub fn get_element_target(subject: &Element) -> Option<DOMString> {
+    if !(subject.is::<HTMLAreaElement>() ||
+        subject.is::<HTMLAnchorElement>() ||
+        subject.is::<HTMLFormElement>())
+    {
+        return None;
+    }
+    if subject.has_attribute(&local_name!("target")) {
+        return Some(
+            subject
+                .get_attribute(&ns!(), &local_name!("target"))
+                .unwrap()
+                .Value(),
+        );
+    }
+    if (document_from_node(subject).base_element())
+        .unwrap()
+        .upcast::<HTMLElement>()
+        .upcast::<Element>()
+        .has_attribute(&local_name!("target"))
+    {
+        return Some(
+            subject
+                .get_attribute(&ns!(), &local_name!("target"))
+                .unwrap()
+                .Value(),
+        );
+    } else {
+        return None;
+    }
+}
+
+/// <https://html.spec.whatwg.org/multipage/links.html#get-an-element's-noopener>
+pub fn get_element_noopener(subject: &Element, target_attribute_value: Option<DOMString>) -> bool {
+    if !(subject.is::<HTMLAreaElement>() ||
+        subject.is::<HTMLAnchorElement>() ||
+        subject.is::<HTMLFormElement>())
+    {
+        return false;
+    }
+    let link_types = subject.get_attribute(&ns!(), &local_name!("rel")).unwrap();
```

Comment:
To avoid panicking, let's do:
```rust
let link_types = match subject.get_attribute(&ns!(), &local_name!("rel")) {
    Some(rel) => rel.Value(),
    None => return false,
};
```